web/bnls.php

```diff
@@ -1,18 +1,27 @@
 <?php
+// Dynamically retrieve the list of online BNLS servers from BNETDocs:
+$url = 'https://bnetdocs.org/servers.json?type_id=4&status=1';
 
-echo "\n";
-// echo "hdx.jbls.org\n";
-// echo "logon.berzerkerjbls.com\n";
-echo "pyro.no-ip.biz\n";
-// echo "alendar.no-ip.org\n";
-// echo "distant.no-ip.org\n";
-// echo "jbls.idiat.com\n";
-// echo "jbls.org\n";
-echo "phix.no-ip.org\n";
-echo "bnls.mattkv.net\n";
-echo "bnls.net\n";
-//echo "ribose.no-ip.org\n";
-//echo "wjlafrance.net\n";
-//echo "7thace.no-ip.org\n";
+header('Content-Type: text/plain;charset=utf-8');
 
-?>
\ No newline at end of file
+function exit_here($message, $code = 500) {
+  if (function_exists('http_response_code')) {
+    http_response_code($code); // introduced in PHP 5.4
+  } else {
+    header($_SERVER['SERVER_PROTOCOL'] . ' ' . $code);
+  }
+  die($message . "\n");
+}
+
+if (!($data = file_get_contents($url))) {
+  exit_here(sprintf('Failed to download server list from: %s', $url));
+}
+
+if (!($json = json_decode($data, true))) {
+  exit_here(sprintf('JSON error #%d while decoding list', json_last_error()));
+}
+
+echo "\n"; // not sure why this is here - @carlbennett
```

Comment:
Not sure why this extra blank line is here. I duplicated it from the static code this script is replacing (see https://github.com/stealthbot/StealthBot/pull/83/files#diff-7d67a09e3b12d93e23a7d59e416c4e1aL2)

Reply:
I think it's used as a check for good data return.
https://github.com/stealthbot/StealthBot/blob/master/trunk/frmChat.frm#L2359-L2393

Reply:
Or not... dunno.